### PR TITLE
Chore: add test for eslint v9

### PR DESCRIPTION
**Pull request by ota-meshi**

### CODE REVIEW

File: tests/lib/utils/ast-utils/extract-expression-references.ts [hunk 1]
```diff
@@ -114,72 +114,96 @@ describe("extractExpressionReferences", () => {
...
+                        test: {
+                            rules: {
+                                test: {
+                                    create(context: Rule.RuleContext) {
+                                        return {
+                                            Program(program: ESTree.Program) {
+                                                const scope =
+                                                    context.sourceCode.getScope(
+                                                        program,
+                                                    )
```
Comment (RunDevelopment) [inline tests/lib/utils/ast-utils/extract-expression-references.ts:132]: We are hitting the character limit hard here. Maybe we could extract the `create` function out into a variable and put it above `const linter = new Linter({ configType: "flat" })`?

File: tests/lib/utils/ast-utils/extract-property-references.ts [hunk 2]
```diff
@@ -254,63 +255,85 @@ describe("extractPropertyReferences", () => {
...
-                        Map: "readonly",
-                        BigInt: "readonly",
-                        window: "readonly",
-                        globalThis: "readonly",
+                    plugins: {
+                        // @ts-expect-error -- ignore type error for eslint v9
+                        test: {
+                            rules: {
+                                test: {
+                                    create(context: Rule.RuleContext) {
```
Comment (RunDevelopment) [inline tests/lib/utils/ast-utils/extract-property-references.ts:267]: Same here.

File: tests/lib/utils/get-usage-of-pattern.ts [hunk 3]
```diff
@@ -262,61 +263,83 @@ describe("getUsageOfPattern", () => {
...
-                        Map: "readonly",
-                        BigInt: "readonly",
-                        window: "readonly",
-                        globalThis: "readonly",
+                    plugins: {
+                        // @ts-expect-error -- ignore type error for eslint v9
+                        test: {
+                            rules: {
+                                test: {
+                                    create(context: Rule.RuleContext) {
```
Comment (RunDevelopment) [inline tests/lib/utils/get-usage-of-pattern.ts:275]: And here.

File: tests/lib/utils/type-tracker/test-utils.ts [hunk 4]
```diff
@@ -26,64 +29,81 @@ const filename = path.join(
...
-                BigInt: "readonly",
-                window: "readonly",
-                globalThis: "readonly",
+            files: ["**/*.*"],
+            plugins: {
+                // @ts-expect-error -- ignore type error for eslint v9
+                test: {
+                    rules: {
+                        test: {
+                            create(context: Rule.RuleContext) {
```
Comment (RunDevelopment) [inline tests/lib/utils/type-tracker/test-utils.ts:42]: And here.

**Review by RunDevelopment — COMMENTED** [review]
I just found a few minor things. Otherwise, this looks good!

**Comment by ota-meshi** [comment]
Thank you for the review! I have changed this PR. Could you please check again?

**Review by RunDevelopment — APPROVED** [review]
Looks good to me! Thank you!

**Pull request merged**